metrics-operator/controllers/common/providers/provider.go

```diff
@@ -20,31 +20,38 @@ type KeptnSLIProvider interface {
 }
 
 // NewProvider is a factory method that chooses the right implementation of KeptnSLIProvider
-func NewProvider(provider string, log logr.Logger, k8sClient client.Client) (KeptnSLIProvider, error) {
-	switch strings.ToLower(provider) {
-	case PrometheusProviderName:
+func NewProvider(providerName string, providerType string, log logr.Logger, k8sClient client.Client) (KeptnSLIProvider, error) {
+	nameToTest := ""
+	if providerType != "" {
+		nameToTest = providerType
+	} else {
+		nameToTest = providerName
+	}
```

Comment:
can we extract this to a helper function? Would look much cleaner

Reply:
done